src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2073,7 +2073,9 @@ uct_ib_mlx5_devx_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
     flags = UCS_PARAM_VALUE(UCT_MD_MKEY_PACK_FIELD, params, flags, FLAGS, 0);
     if (flags & UCT_MD_MKEY_PACK_FLAG_EXPORT) {
 #if HAVE_IBV_DM
-        ucs_assert_always(memh->dm == NULL);
+        if (memh->dm == NULL) {
+            return UCS_ERR_INVALID_PARAM;
```

Comment:
pls also print a nice error message

Reply:
please add more details to the message:
`ucs_error("%s: cannot export memory allocated on the device (address %p length %zu)", uct_ib_device_name(...), ...);`